components/calendar/src/iso.rs

```diff
@@ -383,22 +384,25 @@ impl Iso {
     // Lisp code reference: https://github.com/EdReingold/calendar-code2/blob/1ee51ecfaae6f856b0d7de3e36e9042100b4f424/calendar.l#L1167-L1189
     pub(crate) fn fixed_from_iso(date: IsoDateInner) -> i32 {
         // Calculate days per year
-        let mut fixed: i32 = EPOCH - 1 + 365 * (date.0.year - 1);
+        let mut fixed: i32 = (EPOCH - 1 + 365).saturating_mul(date.0.year.saturating_sub(1));
         // Adjust for leap year logic
-        fixed += quotient(date.0.year - 1, 4) - quotient(date.0.year - 1, 100)
-            + quotient(date.0.year - 1, 400);
+        fixed = fixed.saturating_add(
+            quotient(date.0.year.saturating_sub(1), 4)
+                - quotient(date.0.year.saturating_sub(1), 100)
+                + quotient(date.0.year.saturating_sub(1), 400),
+        );
         // Days of current year
-        fixed += quotient(367 * (date.0.month as i32) - 362, 12);
+        fixed = fixed.saturating_add(quotient(367 * (date.0.month as i32) - 362, 12));
```

Comment:
thought: we don't have to `saturating_add` each time here; we could instead add up into an `offset` variable and then `fixed = fixed.saturating_add(offset)` in the end